internal/blockchain/ethereum/private_transaction_manager.go

```diff
@@ -0,0 +1,262 @@
+		return err
+	}
+	if err := docker.CopyFileToVolume(ctx, volumeName, filepath.Join(tesseraKeyPath, "tm.pub"), tmpath); err != nil {
+		return err
+	}
+	if err := docker.CopyFileToVolume(ctx, volumeName, filepath.Join(tesseraKeyPath, "tm.key"), tmpath); err != nil {
+		return err
+	}
+	return nil
+}
+
+func CreateTesseraEntrypoint(ctx context.Context, outputDirectory, volumeName, memberCount string) error {
+	// only tessera v09 onwards is supported
+	var sb strings.Builder
+	memberCountInt, _ := strconv.Atoi(memberCount)
+	for i := 0; i < memberCountInt; i++ {
+		sb.WriteString(fmt.Sprintf("{\"url\":\"http://member%dtessera:9000\"},", i)) // construct peer list
+	}
+	peerList := strings.TrimSuffix(sb.String(), ",")
+	content := fmt.Sprintf(`export JAVA_OPTS="-Xms128M -Xmx128M"
+DDIR=/data/qdata/dd
+mkdir -p ${DDIR}
+[ -f ${DDIR}/dd/tm.ipc ] && rm -f ${DDIR}/dd/tm.ipc
+cat <<EOF > ${DDIR}/tessera-config-09.json
+	{
+		"useWhiteList": false,
+		"jdbc": {
+			"username": "sa",
+			"password": "",
+			"url": "jdbc:h2:./${DDIR}/db;TRACE_LEVEL_SYSTEM_OUT=0",
+			"autoCreateTables": true
+		},
+		"serverConfigs":[
+			{
+				"app":"ThirdParty",
+				"enabled": true,
+				"serverAddress": "http://$(hostname -i):%s",
+				"communicationType" : "REST"
+			},
+			{
+				"app":"Q2T",
+				"enabled": true,
+				"serverAddress": "http://$(hostname -i):%s",
+				"sslConfig": {
+					"tls": "OFF"
+				},
+				"communicationType" : "REST"
+			},
+			{
+				"app":"P2P",
+				"enabled": true,
+				"serverAddress": "http://$(hostname -i):%s",
+				"sslConfig": {
+					"tls": "OFF"
+			},
+				"communicationType" : "REST"
+			}
+		],
+		"peer": [
+			%s
+		],
+		"keys": {
+		"passwords": [],
+		"keyData": [
+				{
+					"privateKeyPath": "${DDIR}/keystore/tm.key",
+					"publicKeyPath": "${DDIR}/keystore/tm.pub"
+				}
+			]
+		},
+		"alwaysSendTo": [],
+		"bootstrapNode": false,
+		"features": {
+			"enableRemoteKeyValidation": false,
+			"enablePrivacyEnhancements": true
+		}
+	}
+EOF
+/tessera/bin/tessera -configfile ${DDIR}/tessera-config-09.json
+`, TmTpPort, TmQ2tPort, TmP2pPort, peerList)
+	filename := filepath.Join(outputDirectory, entrypoint)
+	file, err := os.OpenFile(filename, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0755)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	_, err = file.WriteString(content)
+	if err != nil {
+		return err
+	}
+	CopyTesseraEntrypointToVolume(ctx, outputDirectory, volumeName)
+	return nil
+}
+
+func CopyTesseraEntrypointToVolume(ctx context.Context, tesseraEntrypointDirectory, volumeName string) error {
+	if err := docker.MkdirInVolume(ctx, volumeName, ""); err != nil {
+		return err
+	}
+	if err := docker.CopyFileToVolume(ctx, volumeName, filepath.Join(tesseraEntrypointDirectory, entrypoint), ""); err != nil {
+		return err
+	}
+	return nil
+}
+
+func CreateQuorumEntrypoint(ctx context.Context, outputDirectory, volumeName, memberIndex string, chainId int) error {
```

Comment:
Ok - so this isn't just `Tessera` private transaction manager support, but also Quorum support - that's significant and interesting. Really needs to be covered in the description etc.

Reply:
Apologize for the mismatch in the naming of the file, starting Quorum has some coupled dependencies to Tessera, and hence some of the variables are common across the two containers.

I'll refactor to make this clearer.

Reply:
Shifted private_transaction_manager.go to a Quorum specific package, and refactor out Quorum docker entry point file creation to its own quorum.go file